arbnode/inbox_tracker.go

```diff
@@ -300,10 +302,12 @@ func (t *InboxTracker) PopulateFeedBacklog(broadcastServer *broadcaster.Broadcas
 			return fmt.Errorf("error getting message %v: %w", seqNum, err)
 		}
 
-		msgResult, err := t.txStreamer.ResultAtCount(seqNum)
 		var blockHash *common.Hash
-		if err == nil {
-			blockHash = &msgResult.BlockHash
+		if seqNum > 0 {
+			msgResult, err := t.txStreamer.ResultAtMessageIndex(seqNum)
```

Comment:
Yes it was wrong :(.
I introduced this issue in this [PR](https://github.com/OffchainLabs/nitro/pull/2240) :sweat:. 
We didn't catch it because we didn't have a test covering PopulateFeedBacklog :grimacing:.

I opened another PR https://github.com/OffchainLabs/nitro/pull/2551 that fixes this issue and creates a test to cover this function.
I will updated this PR once this other one is merged.

Reply:
Merged the other PR and updated this one.